### DISCUSSION TITLE
Adding transaction UI

### DISCUSSION THREAD

**Discussion by RomuloGatto** [ananthakumaran/paisa#63] [Ideas]
The addition of transactions are all manually, editing a text field. Is there any provision to a UI of this creation? On the way of a non-tech-accounting guy being able to use that, and that cli only being able as an expert mode?

**Comment by ananthakumaran** [comment]
I am interested in non-tech people using this tool, which is why a desktop app exists. It's not just a text field, but it has syntax highlighting, error reporting, etc. I briefly thought about adding form based transaction addition, but there are too many issues with that approach, and my belief is, that it will take more time to use the form rather than typing via the editor.

I have been using text editors for too long and might not realize the issue faced by newcomers. If there is any specific issue that prevents non-tech people from using the editor, let me know, and I will try to handle it.

**Reply by RomuloGatto** [reply]
Aside from being a tech guy, I wasn't aware of this ledger code, I found the project and was expecting some UI to add my transactions to that, note learning some accounting way of coding.

`I briefly thought about adding form based transaction addition, but there are too many issues with that approach, and my belief is, that it will take more time to use the form rather than typing via the editor.`
That's exactly what I imagined, that's why I thought about a selector Friendly/Expert, and then the user can use the way it suits better
